core/templates/dev/head/app.js

```diff
@@ -253,10 +259,13 @@ oppia.factory('validatorsService', [
 
 // Service for setting focus. This broadcasts a 'focusOn' event which sets
 // focus to the element in the page with the corresponding focusOn attribute.
-oppia.factory('focusService', ['$rootScope', '$timeout', function($rootScope, $timeout) {
+oppia.factory('focusService', ['deviceInfoService', '$rootScope', '$timeout', function(deviceInfoService, $rootScope, $timeout) {
   var _nextLabelToFocusOn = null;
   return {
     setFocus: function(name) {
+      if (deviceInfoService) {
```

Comment:
Actually, instead of this I do mean adding a separate setFocusIfOnDesktop() function that does what you describe, except that instead of replacing all setFocus calls in the codebase, I would replace them only for the conversation skin (at least for now).